compositor_render/src/scene/transition.rs

```diff
@@ -39,49 +39,64 @@ impl TransitionState {
     pub fn new(
         current_transition: Option<TransitionOptions>,
         previous_transition: Option<TransitionState>,
+        component_props_changed: bool,
+        reset_transition: bool,
         last_pts: Duration,
     ) -> Option<Self> {
+        // Props didn't change therefor there is nothing to transition to
+        let current_transition = match component_props_changed {
```

Comment:
understanding what happens here and when is very hard, technically decision is made based on previous_transition and current_transition, but here you are modifying those values which spreads the actual logic without any structure